src/app/founderOverview/page.tsx

```diff
@@ -1,8 +1,52 @@
+"use client";
+
+import CatchmentTable from "@/components/table";
+import texts from "@/utils/texts";
+import { remult } from "remult";
+import { Founder } from "@/entities/Founder";
+import { useState } from "react";
+import type { ColumnDefinition, TableState } from "@/types/tableTypes";
+
+const foundersRepo = remult.repo(Founder);
+
 export default function FounderOverview() {
+  const columnDefinitions: ColumnDefinition[] = [
+    { title: 'Název', key: 'name' },
+    { title: 'IČO', key: 'ico' },
+    { title: 'Kraj', key: 'city.region.name' },
+    { title: 'Okres', key: 'city.county.name' },
+    { title: 'ORP', key: 'city.orp.name' },
+    { title: 'Počet Škol', key: 'schoolCount' },
+  ];
+
+  const [tableState, setTableState] = useState<TableState>({
```

Comment:
To API pro tabulku se mi moc líbí, jen je IMO zbytečné mít tady state - nevím, jestli je třeba ho mít mimo datagrid. Ideálně bych ho vytvořil a zavřel do datagridu.

Jen bude třeba přidat ty state values do fetchItems jako argumenty, např. takto:
```
const fetchItems = async ({ page, perPage }) => {
 // load data
}
```

A pak časem přidáš i argumenty pro filtrování. Čím víc střev je uzavřených v komponentě, tím líp. A díky tomu můžeš mít tuhle celou stránku klidně jako server komponentu.

Reply:
Jo, rozumím. Souhlasím, že není ideální to mít přímo v page, ale určitě bych to nechtěl dávat přímo do komponenty tabulky. Tam chci nechávat jen logiku konkrétně pro tabulku, jinak bych to musel pokaždé opakovat. Spíš nějakou mezikomponentu, která bude řešit definici té tabulky, data pro tabulku atd.
Ale hlavně jsem si říkal, jak ten next ještě tolik nemám najetý, tak že to budu programovat spíš nějak na první dobrou a pak se nad tím zamyslím, jak by to šlo zoptimalozovat a zrefaktoruju to. Když jsem se to snažil vymyslet na první dobrou, tak jsem byl trošku paralizovaný, abych to udělal dobře a trvá to pak...

Reply:
Však v pohodě, první pokus super, ale tohle je změna na 5 minut - stačí fakt jen přesunout ty stavy dovnitř a při volání fetchItems tam přidat ty parametry. Pokud něco nepřehlížím.

Reply:
Shodou okolností jsem to už vyseparoval do komponenty zvlášť, ještě se pak kouknu, jestli by ten state nešel přesunout níž.